lib/engine/game/g_18_tn.rb

```diff
@@ -107,24 +112,33 @@ def lnr
       end
 
       def upgrades_to?(from, to, special = false)
-        return true if super
-
         # When upgrading from green to brown:
-        #   Memphis (H3) has no label. P label or no label is OK.
-        #   Chattanooga (H15) has C label. Only P label is OK.
-        #   Nashville (F11) has N label. Only P label is OK.
-        from.color == :green && HEX_WITH_P_LABEL.include?(from.hex.name) && to.color == :brown && to.label.to_s == 'P'
+        #   If Memphis (H3), Chattanooga (H15), Nashville (F11)
+        #   only brown P tile (#170) are allowed.
+        return to.name == '170' if from.color == :green && HEX_WITH_P_LABEL.include?(from.hex.name)
+
+        # When upgrading Nashville from yellow to green, only TN2 from green to brown:
+        return to.name == 'TN2' if from.color == :yellow && from.hex.name == 'F11'
+
+        super
       end
 
       def all_potential_upgrades(tile)
-        upgrades = super
+        suggested_upgrades = super
+
+        # Upgrade to green in Nashville (F11) is only allowed with N tile (#TN2)
+        # Tile manifest for yellow standard cities should show N tile (TN1) as an option
+        special_upgrade, updated_upgrades =
```

Comment:
this pattern is strange, why are you returning two arrays here

Reply:
That's true - I can just return nil / updated_upgrades